### PR TITLE
Codespell

### PR DESCRIPTION
**Pull request by hko-s**
testing, testing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,15 +4,16 @@ All notable changes to rpgp will be documented in this file.
 
 ## [0.12.0-alpha.2](https://github.com/rpgp/rpgp/compare/v0.12.0-alpha.1..0.12.0-alpha.2) - 2024-04-07
 
-### ⛰️  Features
```
Comment (dignifiedquire) [inline CHANGELOG.md:7]: the changelog is auto regenerated on release, so not really much point of fixing anything

File: README.md [hunk 2]
```diff
@@ -66,14 +66,14 @@ let key_file = "key.sec.asc";
 let msg_file = "msg.asc";
 
 let key_string = fs::read_to_string("key.sec.asc").unwrap(),
-let (secret_key, _headers) = SignedSecretKey::from_string(&key_string).unwrap();
+let (secret_key, _headers) = SignedSecretKey::from_string( & key_string).unwrap();
```
Comment (dignifiedquire) [inline README.md:69]: this is definitely wrong

File: README.md [hunk 3]
```diff
@@ -125,11 +123,19 @@ for inclusion in this project by you, as defined in the Apache-2.0 license,
 shall be dual licensed as above, without any additional terms or conditions.
 
 [RFC2440]: https://tools.ietf.org/html/rfc2440
+
```
Comment (dignifiedquire) [inline README.md:126]: :/

**Comment by hko-s** [comment]
uhh, i'll have to get my editor under control, clearly :sweat_smile:

File: CHANGELOG.md (hunk 1 above)
Comment (hko-s) [reply CHANGELOG.md:7]: right, will exclude

**Comment by hko-s** [comment]
Phew, that's better.

I'd love to know how to do c8432243f921feba2a16f2dfe38f881147470d9c in a cleaner way. But other than that, this PR seems reasonable to me.

**Comment by hko-s** [comment]
> ... but I wonder if it's nicer to have the config in ci.yml or in a .codespellrc?

I decided I like `.codespellrc` better, and changed the PR.
CI currently fails for unrelated reasons (https://github.com/rpgp/rpgp/issues/348).

I'm still unhappy about this: https://github.com/rpgp/rpgp/pull/341/commits/3ae3f3abc998c32bb2b54f5ffe801f08ac848793
I'd really like to use something similar to an `#[allow(..)]` to tell codespell to ignore that one occurrence, but haven't found any appropriate mechanism in codespell.

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**